src/omnisharp/messageType.ts

```diff
@@ -0,0 +1,178 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { IObserver } from "rx";
+import { PlatformInformation } from "../platform";
+import { Request } from "./requestQueue";
+import * as protocol from './protocol';
+
+export type MessageObserver = IObserver<Message>;
+export enum MessageType {
```

Comment:
I'm a little surprised by this. From our earlier talks, I had imagined a more decentralized approach to defining these message types. That's not to say that this is the wrong approach though.

Reply:
Open to suggestions :)